VideoLocker/src/main/java/org/edx/mobile/view/NavigationFragment.java

```diff
@@ -111,43 +144,43 @@ public void onClick(View v) {
-                    if (environment.getConfig().getEnrollmentConfig().isEnabled()) {
-                        if (!(act instanceof FindCoursesActivity)) {
-                            environment.getRouter().showFindCourses(act);
-
-                            //Finish need not be called if the current activity is MyCourseListing
-                            // as on returning back from FindCourses,
-                            // the student should be returned to the MyCourses screen
-                            if (!(act instanceof MyCoursesListActivity)) {
-                                act.finish();
-                            }
-                        }
-                    } else {
-                        //Show the dialog only if the activity is started. This is to avoid Illegal state
-                        //exceptions if the dialog fragment tries to show even if the application is not in foreground
-                        if (isAdded() && isVisible()) {
-                            FindCoursesDialogFragment findCoursesFragment = new FindCoursesDialogFragment();
-                            findCoursesFragment.setStyle(DialogFragment.STYLE_NORMAL,
-                                    android.R.style.Theme_Black_NoTitleBar_Fullscreen);
-                            findCoursesFragment.setCancelable(false);
-                            findCoursesFragment.show(getFragmentManager(), "dialog-find-courses");
+        TextView tvFindCourses = (TextView) layout.findViewById(R.id.drawer_option_find_courses);
+        tvFindCourses.setVisibility(View.GONE);
+        tvFindCourses.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                try {
+                    ISegment segIO = environment.getSegment();
+                    segIO.trackUserFindsCourses();
+                } catch (Exception e) {
+                    logger.error(e);
+                }
+                Activity act = getActivity();
+                ((BaseFragmentActivity) act).closeDrawer();
+                if (environment.getConfig().getEnrollmentConfig().isEnabled()) {
+                    if (!(act instanceof FindCoursesActivity)) {
+                        environment.getRouter().showFindCourses(act);
+
+                        //Finish need not be called if the current activity is MyCourseListing
+                        // as on returning back from FindCourses,
+                        // the student should be returned to the MyCourses screen
+                        if (!(act instanceof MyCoursesListActivity)) {
+                            act.finish();
                         }
                     }
+                } else {
+                    //Show the dialog only if the activity is started. This is to avoid Illegal state
+                    //exceptions if the dialog fragment tries to show even if the application is not in foreground
+                    if (isAdded() && isVisible()) {
```

Comment:
Note that `isVisible()` does not indicate that the activity is started. For that `isResumed()` could be used I guess. Although, if the user is interacting with the fragment, then that does indicate that the activity is in the foreground.